From: Owl Linux glibc-2.3.2-suse-resolv-response-length.diff

Warning: This patch can screw up applications that expect the vanilla
behavior. See: http://webui.sourcelabs.com/samba/issues/4014

diff -Naur glibc-2.8-20080929.orig/resolv/res_send.c glibc-2.8-20080929/resolv/res_send.c
--- glibc-2.8-20080929.orig/resolv/res_send.c	2007-08-22 04:02:38.000000000 +0000
+++ glibc-2.8-20080929/resolv/res_send.c	2008-10-17 03:17:19.000000000 +0000
@@ -384,6 +384,17 @@
 					needclose++;
 					break;
 				}
+				/*
+				 * Dirty fix - avoid read buffer overruns in
+				 * applications that naively assume the length
+				 * returned by res_* is always less than or
+				 * equal the answer buffer size.
+				 *
+				 * Simply truncating the answer here surely
+				 * beats fixing all calls of res_* in all
+				 * applications.
+				 */
+				resplen = anssiz;
 			}
 		if (needclose)
 			__res_iclose(statp, false);
